cl/corpus_importer/import_columbia/columbia_utils.py

```diff
@@ -0,0 +1,326 @@
+    "memorandum dated",
+    "mandamus denied on mandate",
+    "updated",
+    "date of judgment entered",
+    "released and journalized",
+    "submitted on",
+    "case assigned",
+    "opinion circulated for comment",
+    "submitted on rehearing",
+    "united states supreme court dismissed appeal",
+    "answered",
+    "reconsideration granted in part and as amended",
+    "as amended on denial of rehearing",
+    "reassigned",
+    "as amended",
+    "as corrected",
+    "writ allowed",
+    "released",
+    "application for leave to appeal filed",
+    "affirmed on appeal reversed and remanded",
+    "as corrected",
+    "withdrawn substituted and refiled",
+    "answered",
+    "released",
+    "as modified and ordered published",
+    "remanded",
+    "concurring opinion added",
+    "decision and journal entry dated",
+    "memorandum filed",
+    "as modified",
+]
+
+
+def convert_columbia_html(text):
+    """Convert xml tags to html tags
+    :param text: Text to convert to html
+    """
+    conversions = [
+        ("italic", "em"),
+        ("block_quote", "blockquote"),
+        ("bold", "strong"),
+        ("underline", "u"),
+        ("strikethrough", "strike"),
+        ("superscript", "sup"),
+        ("subscript", "sub"),
+        ("heading", "h3"),
+        ("table", "pre"),
+    ]
+
+    for pattern, replacement in conversions:
+        text = re.sub(f"<{pattern}>", f"<{replacement}>", text)
+        text = re.sub(f"</{pattern}>", f"</{replacement}>", text)
+
+    # Make nice paragraphs. This replaces double newlines with paragraphs, then
+    # nests paragraphs inside blockquotes, rather than vice versa. The former
+    # looks good. The latter is bad.
+    text = f"<p>{text}</p>"
+    text = re.sub(r"</blockquote>\s*<blockquote>", "\n\n", text)
+    text = re.sub("\n\n", "</p>\n<p>", text)
+    text = re.sub("\n  ", "</p>\n<p>", text)
+    text = re.sub(r"<p>\s*<blockquote>", "<blockquote><p>", text, re.M)
+    text = re.sub("</blockquote></p>", "</p></blockquote>", text, re.M)
+
+    return text
+
+
+def convert_columbia_opinion(text: str, opinion_index: int) -> str:
+    """Convert xml tags to html tags and process additional data from opinions
+    like footnotes,
+    :param text: Text to convert to html
+    :param opinion_index: opinion index from a list of all opinions
+    :return: converted text
+    """
+
+    text = convert_columbia_html(text)
+
+    # grayed-out page numbers
+    text = re.sub("<page_number>", ' <span class="star-pagination">*', text)
+    text = re.sub("</page_number>", "</span> ", text)
+
+    # footnotes
+    foot_references = re.findall(
+        "<footnote_reference>.*?</footnote_reference>", text
+    )
+
+    for ref in foot_references:
+        if (match := re.search(r"[\*\d]+", ref)) is not None:
+            f_num = match.group()
+        elif (match := re.search(r"\[fn(.+)\]", ref)) is not None:
+            f_num = match.group(1)
+        else:
+            f_num = None
```

Comment:
was there an issue with just reusing this code?
`convert_columbia_html`

Reply:
It's because the old code did not bind the footnotes correctly, i fixed that so the footnotes for each possible opinion are independent and the links works fine now.